ui/app/components/clients/history.js

```diff
@@ -89,48 +90,68 @@ export default class History extends Component {
-      if (isAfter(versionDate, startTime) && isBefore(versionDate, endTime)) {
-        // push the minor version number e.g. 9
-        let upgradeVersionArray = object.id.split('.');
-        newArray.push(upgradeVersionArray[1]);
-      }
+  get upgradeDuringActivity() {
+    const activityStart = new Date(this.getActivityResponse.startTime);
+    const activityEnd = new Date(this.getActivityResponse.endTime);
+    const upgradesWithinData = this.upgradeVersionHistory.filter((upgrade) => {
+      // will timezones be an issue because new Date defaults to current timezone?
+      let upgradeDate = new Date(upgrade.timestampInstalled);
+      return isAfter(upgradeDate, activityStart) && isBefore(upgradeDate, activityEnd);
     });
-    return newArray; // return [9,9,10] shows all the minor numbers that fall within the range of the returned API response.
+    // return all upgrades that happened within date range of queried activity
+    return upgradesWithinData.length === 0 ? null : upgradesWithinData;
   }
 
-  get versionUpdateText() {
-    // count how many times a minor number occurs inside the array of minors within the API range.
-    // occurrences returns: {9: 2, 10: 1}. key is the minor version and value is the count.
-    let occurrences = this.allMinorVersionsInRange.reduce(function (acc, curr) {
-      return acc[curr] ? ++acc[curr] : (acc[curr] = 1), acc;
-    }, {});
-    // if count of minor 9 is greater than one and there are no occurrences of minor 10, show the minor 9 specific warning
-    if (occurrences[9] >= 1 && !occurrences[10]) {
-      return `Vault was upgraded to ${this.latestUpgradeData.id} during this time range. How we count clients changed in 1.9, so keep that in mind when looking at the data below. `;
-    } else if (occurrences[10] >= 1 && !occurrences[9]) {
-      // show the minor 10 specific warning
-      return `Vault was upgraded to ${this.latestUpgradeData.id} during this time range. We added monthly breakdowns for 1.10 data only, so keep that in mind when looking at the data below. `;
-    } else if (occurrences[10] >= 1 && occurrences[9] >= 1) {
-      // you have both 9 and 10 as minors. Show generic warning.
-      return `Vault was upgraded to 1.9 and 1.10 during this time range. How we count clients changed in 1.9 and we added monthly breakdowns for 1.10 data only. Keep this in mind when looking at the data below. `;
+  get upgradeVersionAndDate() {
+    if (this.upgradeDuringActivity.length === 2) {
+      return 'Vault was upgraded to 1.9 and 1.10 during this time range.';
+    } else {
+      let upgrade = this.upgradeDuringActivity[0];
+      return `Vault was upgrade to ${upgrade.id} on ${dateFormat(
+        [upgrade.timestampInstalled, 'MMM d, yyyy'],
+        { isFormatted: true }
+      )}.`;
+    }
+  }
+
+  get versionSpecificText() {
+    if (this.upgradeDuringActivity.length === 1) {
+      let version = this.upgradeDuringActivity[0].id;
```

Comment:
`version.match('1.9') will match regardless of what the patch release is, so both '1.9.1' or '1.9.5' will hit the correct `if` block